Repository/Api/EditableRepository.php

```diff
@@ -39,4 +39,13 @@
      * @throws UnsupportedLanguageException If the language is not supported.
      */
     public function move($sourceQuery, $targetPath, $language = 'glob');
+
+    /**
+     * Change the position of a node relative to its siblings.
+     *
+     * @param string $sourcePath
+     * @param int $position
+     */
+    public function reposition($sourcePath, $position);
```

Comment:
> with a bit of thinking, i see why you chose the name "reposition"

I can also see how that is confusing. I think I just wanted to distance myself from "reorder" with my subjective association with reordering using names rather than indexes.

> does puli know this concept?

No.

> its closer to how a human thinks when moving the node

But involves more logic for machines. With an absolute index you do not have to worry about special logic for the end/beginning node.